src/Microsoft.DotNet.CodeFormatting/Analyzers/OptimizeNamespaceImportsAnalyzer.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.DotNet.CodeFormatting.Analyzers
+{
+    [Export(typeof(DiagnosticAnalyzer))]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class OptimizeNamespaceImportsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = AnalyzerIds.OptimizeNamespaceImports;
+        private static DiagnosticDescriptor s_rule = new DiagnosticDescriptor(DiagnosticId,
+                                                                            ResourceHelper.MakeLocalizableString(nameof(Resources.OptimizeNamespaceImportsAnalyzer_Title)),
+                                                                            ResourceHelper.MakeLocalizableString(nameof(Resources.OptimizeNamespaceImportsAnalyzer_MessageFormat)),
+                                                                            "Style",
+                                                                            DiagnosticSeverity.Warning,
+                                                                            true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(s_rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSemanticModelAction(semanticModelAnalysisContext =>
+            {
+                SyntaxNode root;
```

Comment:
Let us let @jaredpar be the authority on whether he wants to combine-declaration-with-initialization in this code base. You know how I feel about it ;-)